client/finality-grandpa/src/communication/mod.rs

```diff
@@ -67,9 +67,23 @@ mod periodic;
 #[cfg(test)]
 pub(crate) mod tests;
 
-/// Name of the notifications protocol used by Grandpa. Must be registered towards the networking
-/// in order for Grandpa to properly function.
-pub const GRANDPA_PROTOCOL_NAME: &'static str = "/paritytech/grandpa/1";
+pub(crate) mod grandpa_protocol_name {
+	const NAME: &'static str = "/grandpa/1";
+	/// Old names for the notifications protocol, used for backward compatibility.
+	pub const LEGACY_NAMES: [&'static str; 1] = ["/paritytech/grandpa/1"];
+
+	/// Name of the notifications protocol used by Grandpa.
+	/// Must be registered towards the networking in order for Grandpa to properly function.
+	pub fn with_prefix(prefix: &str) -> String {
+		format!("{}{}", prefix, NAME)
+	}
+
+	#[cfg(test)]
+	/// Protocol name used in tests.
+	pub fn test_name() -> String {
+		format!("/test{}", NAME)
+	}
```

Comment:
I think that the test code should not pull the name from here. What happens in the tests is specific to the tests and shouldn't infest the non-test code.